symfony/plugins/orangehrmAttendancePlugin/Api/AttendanceRecordAPI.php

```diff
@@ -0,0 +1,339 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Attendance\Api;
+
+use DateTime;
+use DateTimeZone;
+use Exception;
+use OrangeHRM\Attendance\Api\Model\AttendanceRecordModel;
+use OrangeHRM\Attendance\Exception\AttendanceServiceException;
+use OrangeHRM\Attendance\Traits\Service\AttendanceServiceTrait;
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Exception\BadRequestException;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\ResourceEndpoint;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Service\DateTimeHelperService;
+use OrangeHRM\Core\Traits\Auth\AuthUserTrait;
+use OrangeHRM\Core\Traits\Service\DateTimeHelperTrait;
+use OrangeHRM\Core\Traits\UserRoleManagerTrait;
+use OrangeHRM\Entity\AttendanceRecord;
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\Entity\WorkflowStateMachine;
+
+class AttendanceRecordAPI extends Endpoint implements ResourceEndpoint
+{
+    use AttendanceServiceTrait;
+    use AuthUserTrait;
+    use UserRoleManagerTrait;
+    use DateTimeHelperTrait;
+
+    public const PARAMETER_PUNCH_IN_DATE = 'punchInDate';
+    public const PARAMETER_PUNCH_IN_TIME = 'punchInTime';
+    public const PARAMETER_PUNCH_IN_NOTE = 'punchInNote';
+    public const PARAMETER_PUNCH_OUT_DATE = 'punchOutDate';
+    public const PARAMETER_PUNCH_OUT_TIME = 'punchOutTime';
+    public const PARAMETER_PUNCH_OUT_NOTE = 'punchOutNote';
```

Comment:
@super-chama can we go with below JSON structure for edit records?
```json
{
  "punchInDate": "",
  "punchInTime": "",
  "punchInNote": "",
  "punchOutDate": "",
  "punchOutTime": "",
  "punchOutNote": "",
}
```

Reply:
should be good